lib/mix/lib/mix/project.ex

```diff
@@ -430,7 +430,7 @@ defmodule Mix.Project do
   """
   @spec build_path(keyword) :: Path.t()
   def build_path(config \\ config()) do
-    config[:env_path] || env_path(config)
+    System.get_env("MIX_BUILD_PATH") || config[:env_path] || env_path(config)
```

Comment:
We document all environment variables in lib/mix.ex. We need to add this one there.